src/EventStore.Core/Synchronization/IExclusiveLock.cs

```diff
@@ -0,0 +1,6 @@
+namespace EventStore.Core.Synchronization;
+
+public interface IExclusiveLock {
```

Comment:
I could be persuaded, but is this abstraction helping - I suspect that the different implementations would have have quite different properties and not really be substitutable i.e. the consumer would likely be relying on it being a semaphoreslim in particular

Reply:
yup, good point, i'll remove the interface

Reply:
fixed

Reply:
fixed in: d39ec1679772c86b11779b9e5c5ef07c4e0ff630